lib/statemanager.js

```diff
@@ -883,15 +883,20 @@ StateManager.prototype.createFakeTransactionWithCorrectNonce = function(rawTx, f
 
     // If we're calling a contract, check to make sure the address specified is a contract address
     if (_transactionIsContractCall(rawTx)) {
-      self.getCode(to.hex(rawTx.to), 'latest', function(err, code) {
-        if (err) {
-          callback(err);
-        } else if (code === '0x0') {
-          callback(new TXRejectedError(`Attempting to run transaction which calls a contract function, but recipient address ${to.hex(rawTx.to)} is not a contract address`))
-        } else {
-          callback(null, tx)
-        }
-      });
+      // Contract just being deployed, code yet to exist at adress
+      if (!rawTx.to || rawTx.to === '0x0' || rawTx.to === '0x' || rawTx.to === '0') {
```

Comment:
FWIW, I think contract deployment only happens when the `to` field is completely missing. It's completely valid to send a transaction with data to address zero.

Reply:
That said, `to` is a DATA field (per the [RPC wiki](https://github.com/ethereum/wiki/wiki/JSON-RPC#eth_sendtransaction)), so the zero-length value of `0x` should probably also be considered a contract deployment.